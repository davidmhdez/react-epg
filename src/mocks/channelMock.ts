import Channel, {ChannelEvent} from "../domain/entity/Channel.ts";
import ProgramGuide from "../domain/entity/ProgramGuide.ts";

export const channelEventMock: ChannelEvent = {
    id: "11",
    name: "Test event",
    duration: "01:30:00",
    dateBegin: new Date(2024, 6, 25, 14, 0, 0),
    description: "Test description",
    durationInMinutes: 90,
    dateEnd: new Date(2024, 6, 25, 15, 30, 0),
}

export const channelMock: Channel = {
    id: "1",
    name: "Test channel",
    number: "3",
    image: "https://www.testimage.com/assets/image.png",
    events: [
        channelEventMock
    ]
}

export const programGuideMock: ProgramGuide = {
    from: new Date(2024, 6, 25, 14, 0, 0),
    end: new Date(2024, 6, 25, 23, 59, 59),
    channels: [
        channelMock
    ]
}