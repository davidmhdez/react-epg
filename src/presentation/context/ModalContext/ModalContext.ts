import {createContext} from "react";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type ModalContextValues = {
    isOpen: boolean
    openModal: () => void
    closeModal: () => void
    selectedChannelEvent: ChannelEvent | null
    selectChannelEvent: (channelEvent: ChannelEvent) => void
}

export const initialModalContextValues = {
    isOpen: false,
    selectedChannelEvent: null,
    closeModal: () => {},
    openModal: () => {},
    selectChannelEvent: () => {},
}

const ModalContext = createContext<ModalContextValues>(initialModalContextValues)

export default ModalContext