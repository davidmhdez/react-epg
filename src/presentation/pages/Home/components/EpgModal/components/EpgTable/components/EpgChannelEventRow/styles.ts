import styled from "styled-components";

export const EpgChannelEventRowWrapper = styled.div`
    display: flex;
    align-items: center;
    position: relative;
`;

export const EpgChannelEventList = styled.div<{$maxWidth: number}>`
    position: relative;
    display: flex;
    border: 1px solid #bebebe;
    background-color: #8D8D8D86;
    max-width: ${({$maxWidth}) => $maxWidth}px;
    overflow: hidden;
`;