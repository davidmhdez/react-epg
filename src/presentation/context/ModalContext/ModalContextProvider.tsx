import React, {ReactNode, useState} from "react";
import ModalContext from "./ModalContext.ts";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type Props = {
    children: ReactNode
}

const ModalContextProvider: React.FC<Props> = ({children}) => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedChannelEvent, setSelectedChannelEvent] = useState<ChannelEvent | null>(null)
    const openModal = () => {
        setIsOpen(true);
        setSelectedChannelEvent(null);
    }
    const closeModal = () => {
        setIsOpen(false);
        setSelectedChannelEvent(null);
    }

    const selectChannelEvent = (channelEvent: ChannelEvent) => setSelectedChannelEvent(channelEvent);

    return (
        <ModalContext.Provider value={{
            isOpen,
            selectedChannelEvent,
            selectChannelEvent,
            openModal,
            closeModal
        }}>
            {children}
        </ModalContext.Provider>
    );
};

export default ModalContextProvider;
