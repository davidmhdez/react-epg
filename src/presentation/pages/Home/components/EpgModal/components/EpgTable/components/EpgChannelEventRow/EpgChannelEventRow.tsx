import React from 'react';
import Channel from "../../../../../../../../../domain/entity/Channel.ts";
import {EpgChannelEventList, EpgChannelEventRowWrapper} from "./styles.ts";
import EpgChannelEventCol from "./components/EpgChannelEventCol";
import {getDimensions} from "../../../../helpers.ts";
import ChannelCard from "./components/ChannelCard";

type Props = {
    channel: Channel
    programGuideBegin: Date
    maxWidth: number
}

const EpgChannelEventRow: React.FC<Props> = ({channel, programGuideBegin, maxWidth}) => {
    return (
        <EpgChannelEventRowWrapper>
            <ChannelCard
                channel={channel}
            />
            <EpgChannelEventList $maxWidth={maxWidth}>
                {channel.events.map((channelEvent, index) => {
                    const { width, position } = getDimensions(programGuideBegin, channelEvent.dateBegin, channelEvent.durationInMinutes)

                    return(
                        <EpgChannelEventCol
                            channelEvent={channelEvent}
                            key={channelEvent.id}
                            position={position}
                            width={width}
                            index={index}
                        />
                    )
                })}
            </EpgChannelEventList>
        </EpgChannelEventRowWrapper>
    );
};

export default EpgChannelEventRow;
