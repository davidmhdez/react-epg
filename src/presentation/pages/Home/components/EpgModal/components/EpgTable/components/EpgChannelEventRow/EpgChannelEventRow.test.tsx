import {render} from "@testing-library/react";
import ModalContext, {initialModalContextValues} from "../../../../../../../../context/ModalContext/ModalContext.ts";
import EpgChannelEventRow from "./EpgChannelEventRow.tsx";
import {channelMock, programGuideMock} from "../../../../../../../../../mocks/channelMock.ts";

describe('EpgChannelEventRow test', () => {
    test('should match with snapshot', () => {
        const { container } = render(
            <ModalContext.Provider value={initialModalContextValues}>
                <EpgChannelEventRow
                    channel={channelMock}
                    maxWidth={1200}
                    programGuideBegin={programGuideMock.from}
                />
            </ModalContext.Provider>
        )
        expect(container).toMatchSnapshot()
    })
})