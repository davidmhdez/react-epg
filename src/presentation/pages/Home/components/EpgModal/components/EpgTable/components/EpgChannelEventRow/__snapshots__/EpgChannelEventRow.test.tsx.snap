// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpgChannelEventRow test should match with snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB joXNub"
  >
    <div
      class="sc-iGgVNO ZsAZH"
    >
      <div
        class="sc-gsFSjX elKSpN"
      >
        <p
          class="sc-kAycRU bFxGqS"
        >
          3
        </p>
        <img
          alt="Test channel logo"
          class="sc-imWZod bJflWt"
          src="https://www.testimage.com/assets/image.png"
        />
      </div>
    </div>
    <div
      class="sc-gEvDqW efsRrp"
    >
      <div
        class="sc-eqUzNf nJUDT"
        data-testid="11"
      >
        <h6
          class="sc-fqkwJk fGsIES"
          title="Test event"
        >
          Test event
        </h6>
        <p
          class="sc-dcJtft etZvKb"
          title="14:00 - 15:30"
        >
          14:00 - 15:30
        </p>
      </div>
    </div>
  </div>
</div>
`;
